lib/oli/delivery/sections.ex

```diff
@@ -1028,6 +1034,36 @@ defmodule Oli.Delivery.Sections do
     end)
   end
 
+  # Returns a mapset of two element tuples of the form {source_resource_id, target_resource_id}
+  # representing the relates_to relationship between pages.
+  defp get_relates_to(publication_ids) do
+    joined_publication_ids = Enum.join(publication_ids, ",")
+    page_type_id = Oli.Resources.ResourceType.get_id_by_type("page")
+
+    sql = """
+    select
+      rev.resource_id, rev.relates_to
+    from published_resources as mapping
+    join revisions as rev
+    on mapping.revision_id = rev.id
+    where rev.resource_type_id = #{page_type_id} and array_length(rev.relates_to, 1) > 0 and mapping.publication_id IN (#{joined_publication_ids})
+    """
```

Comment:
Is there a reason why we're using raw SQL instead of Ecto queries?

Reply:
I couldn't do this in an Ecto query due to the usage of `array_length`.  I guess I could have used just a fragment for that one part.